### PR TITLE
intersect-resources: Fix incorrect pre-layout assumption

### PR DESCRIPTION
**Pull request by choumx**
Related to #25428.

More closely mimic the classical behavior of "explicit measure before layout" by checking if there's an unprocessed, "premeasured rect" from a more recent intersection callback. 

Noticed this potential issue in examples/everything.amp.html in an `amp-ad > amp-analytics` element that starts with size 1x1 and then gets hidden.

### CODE REVIEW

File: src/service/resources-impl.js [hunk 1]
```diff
@@ -1420,14 +1420,28 @@ export class ResourcesImpl {
...
-        if (!this.intersectionObserver_) {
-          task.resource.measure();
+        const {resource} = task;
+
+        let stillDisplayed = true;
+        if (this.intersectionObserver_) {
+          // With IntersectionObserver, peek at the premeasured rect to see
+          // if the resource is still displayed (has a non-zero size).
+          // The premeasured rect is most analogous to an immediate measure.
+          if (resource.hasBeenPremeasured()) {
```
Comment (samouri) [inline src/service/resources-impl.js:1430]: Am I right in thinking this is to catch the case where the IO has fired and inserted the premeasuredRect but the `discoverWork` block of the pass hasn't happened yet?

File: test/unit/test-resources.js [hunk 2]
```diff
@@ -916,18 +913,54 @@ describes.realWin('Resources discoverWork', {amp: true}, (env) => {
...
+
+      resources.scheduleLayoutOrPreload(resource1, /* layout */ true);
+      resources.work_();
+
+      expect(resources.exec_.getSize()).to.equal(0);
+      expect(resource1.isDisplayed).to.be.calledWith(
+        /* usePremeasuredRect */ true
+      );
+      expect(resource1.layoutCanceled).to.be.calledOnce;
+      expect(resource1.getState()).to.equal(ResourceState.LAYOUT_SCHEDULED);
```
Comment (samouri) [inline test/unit/test-resources.js:950]: doesn't `resource.layoutCancelled()`  set the resource state to either READY_FOR_LAYOUT or NOT_LAID_OUT?

File: src/service/resources-impl.js (hunk 1 above)
Comment (choumx) [reply src/service/resources-impl.js:1430]: Precisely!

File: test/unit/test-resources.js (hunk 2 above)
Comment (choumx) [reply test/unit/test-resources.js:950]: It does, but I stubbed it so the function becomes a no-op.

Changed to spy and updated this state. Good eye.

**Pull request merged**